src/python/pants/jvm/resolve/lockfile_metadata.py

```diff
@@ -24,6 +24,11 @@ class InvalidJVMLockfileReason(Enum):
     REQUIREMENTS_MISMATCH = "requirements_mismatch"
 
 
+class LockfileContext(Enum):
+    USER = "user"
+    TOOL = "tool"
```

Comment:
Might want to move this to a more generic place like `core/util_rules`. I think we'll need this for Python too.